src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs

```diff
@@ -340,12 +340,32 @@ protected virtual void Dispose(bool disposing)
             {
                 if (disposing)
                 {
+                    CleanupPlugins();
                 }
 
                 this.disposed = true;
             }
         }
 
+        private void CleanupPlugins()
+        {
+            EqtTrace.Info("DataCollectionManager.CleanupPlugins: CleanupPlugins called");
+
+            if (!this.isDataCollectionEnabled)
+            {
+                return;
+            }
+
+            if (EqtTrace.IsVerboseEnabled)
+            {
+                EqtTrace.Verbose("DataCollectionManager.CleanupPlugins: Cleaning up {0} plugins", this.RunDataCollectors.Count);
+            }
+
+            RemoveDataCollectors(new List<DataCollectorInformation>(this.RunDataCollectors.Values));
+
+            EqtTrace.Info("DataCollectionManager.CleanupPlugins: CleanupPlugins finished");
```

Comment:
>EqtTrace.Info [](start = 12, length = 13)

why not here then

Reply:
This is compile time string. already created.